tests/test_cli_compile.py

```diff
@@ -3001,6 +3001,37 @@ def test_cli_compile_strip_extras(runner, make_package, make_sdist, tmpdir):
     assert "[more]" not in out.stderr
 
 
+def test_cli_compile_all_extras_with_multiple_packages(
+    runner, make_package, make_sdist, tmpdir
+):
+    """
+    Assures that ``--all-extras`` works when multiple sources are specified.
+    """
+    test_package_1 = make_package(
+        "test_package_1",
+        version="0.1",
+        extras_require={"more": []},
+    )
+    test_package_2 = make_package(
+        "test_package_2",
+        version="0.1",
+        extras_require={"more": []},
+    )
+
+    out = runner.invoke(
+        cli,
+        [
+            "--all-extras",
+            "--output-file",
+            "requirements.txt",
```

Comment:
Could you redirect this to standard output and check it, to make sure?

Reply:
Sorry about the late reply. I printed stdout and stderr now and got:
```
WARNING: the legacy dependency resolver is deprecated and will be removed in future versions of pip-tools.
WARNING: --strip-extras is becoming the default in version 8.0.0. To silence this warning, either use --strip-extras to opt into the new default or use --no-strip-extras to retain the existing behavior.
#
# This file is autogenerated by pip-compile with Python 3.11
# by the following command:
#
#    pip-compile --all-extras --output-file=requirements.txt /tmp/pytest-of-user/pytest-8/popen-gw16/packages0/test_package_1/0.1/setup.py /tmp/pytest-of-user/pytest-8/popen-gw16/packages1/test_package_2/0.1/setup.py
#
```

Reply:
Could you write an assert statement to ensure the header contains `--all-extras` and both of the files?

Reply:
Yes, no problem. I have pushed a fixup with the asserts now. I also tested that they fail if I assert on a different string, such as `test_package_3/0.1/setup.py`.